File: Add Code Here/PYTHON/stack.py
class Node:
	def __init__(self, value):
		self.value = value
		self.next = None


class Stack:
	def __init__(self):
		self.head = Node("head")
		self.size = 0

	def __str__(self):
		cur = self.head.next
		out = ""
		while cur:
			out += str(cur.value) + "->"
			cur = cur.next
		return out[:-2]

	def getSize(self):
		return self.size

	def isEmpty(self):
		return self.size == 0

	def peek(self):
		if self.isEmpty():
			raise Exception("Peeking from an empty stack")
		return self.head.next.value

	def push(self, value):
		node = Node(value)
		node.next = self.head.next
		self.head.next = node
		self.size += 1

	def pop(self):
		if self.isEmpty():
			raise Exception("Popping from an empty stack")
		remove = self.head.next
		self.head.next = self.head.next.next
		self.size -= 1
		return remove.value

if __name__ == "__main__":
	stack = Stack()
	for i in range(1, 11):
		stack.push(i)
	print(f"Stack: {stack}")

	for _ in range(1, 6):
		remove = stack.pop()
		print(f"Pop: {remove}")
	print(f"Stack: {stack}")

File: Add Code Here/C++/Stack.cpp
#include <iostream>
#include <stack>

using namespace std;

int main() {
    stack<int> myStack;

    if (myStack.empty()) {
        cout << "The stack is empty." << endl;
    }

    myStack.push(10);
    myStack.push(20);
    myStack.push(30);

    cout << "Size of the stack: " << myStack.size() << endl;

    cout << "Top element of the stack: " << myStack.top() << endl;

    myStack.pop();

    cout << "Popped an element from the stack." << endl;

    cout << "Top element of the stack: " << myStack.top() << endl;

    if (!myStack.empty()) {
        cout << "The stack is not empty." << endl;
    }

    myStack.push(40);
    myStack.push(50);

    cout << "Size of the stack: " << myStack.size() << endl;

    cout << "Stack elements: ";
    while (!myStack.empty()) {
        cout << myStack.top() << " ";
        myStack.pop();
    }
    cout << endl;

    if (myStack.empty()) {
        cout << "The stack is empty again." << endl;
    }

    return 0;
}
